### PR TITLE
Get refresh button working

### PR DESCRIPTION
**Pull request by robertoallende**

### CODE REVIEW

File: app/src/main/java/com/sportsworld/learnsharedflow/MainActivity.kt [hunk 1]
```diff
@@ -11,7 +11,10 @@ import androidx.compose.runtime.*
 import androidx.compose.foundation.lazy.*
 import androidx.compose.material.*
 import com.sportsworld.sportcomponentlib.DisplayScreenSetup
-
+import com.sportsworld.sportcomponentlib.RefreshButtonLib
+import com.sportsworld.sportcomponentlib.SportFeatureViewModelLib
+import com.sportsworld.sportcomponentlib.RefreshButtonLib
+import com.sportsworld.sportcomponentlib.ui.icons.RefreshIcon
```
Comment (jfsugrue) [inline app/src/main/java/com/sportsworld/learnsharedflow/MainActivity.kt:17]: Breaks here for me?

**Comment by robertoallende** [comment]
Good job. It's great you're using Compose and Flow.

We would need a few changes:
- Can we fix the indentation style on [build.gradle](https://github.com/robertoallende/Sports/blob/get_refresh_button_working/app/build.gradle) please?
- It's great to use new libraries but they don't have to be beta nor alpha, right now we're using '1.1.0-beta01' for compose. Can we fix that please?
- Can we remove defaults test please?

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt [hunk 2]
```diff
@@ -30,46 +26,74 @@ fun TopBarLib() {
...
+    val getNewSportOnClick: () -> Unit = {
+        scope.launch {
+            try {
 
-    LaunchedEffect(key1 = Unit) {
-        sharedFlow.collect {
-            messages.add(0, it)
-            // Todo: Learn more about recomposition so I can find a better way to achieve this
-            messages.removeRange(1,messages.size)
+                // try and collect the value from the flow and then print it out
```
Comment (jfsugrue) [inline sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt:44]: Do we want to add this code here? Pros/Cons?

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt [hunk 3]
```diff
@@ -30,46 +26,74 @@ fun TopBarLib() {
...
+                println("Coroutine to get current list of sports cancelled")
+            }
         }
     }
-    LazyColumn {
-        items(messages) {
-            val sport = it[0].name
-            val description = it[0].description
+
+    RefreshIcon(onClick = getNewSportOnClick) {
```
Comment (jfsugrue) [inline sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt]: This is not building for me here

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/SportFeatureViewModelLib.kt [hunk 4]
```diff
@@ -2,29 +2,26 @@ package com.sportsworld.sportcomponentlib
...
+    val uiState: StateFlow<List<Sport>> = _uiState
 
     init {
-        sharedFlowInit()
+        // allow us to populate the composable initially
+        loadNews()
     }
 
-    fun sharedFlowInit() {
+    fun loadNews() {
```
Comment (jfsugrue) [inline sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/SportFeatureViewModelLib.kt]: News?

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt (hunk 2 above)
Comment (Katie-Marie) [reply sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt:44]: Yes, this sort of processing should be done in the ViewModel. In the improved branch _get_refresh_button_working_ this code doesn't exist. Initially I was using SharedFlow but by using the other Flow API, StateFlow, I could easily access the last value emitted by doing state.value

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/SportFeatureViewModelLib.kt (hunk 4 above)
Comment (Katie-Marie) [reply sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/SportFeatureViewModelLib.kt]: Changed to LoadNewSports

**Comment by Katie-Marie** [comment]
Can we fix the indentation style on [build.gradle] please?

- Hopefully selecting Code -> Reformat File -> Code Cleanup has fixed the indentation style and applied it to the build.gradle

It's great to use new libraries but they don't have to be beta nor alpha, right now we're using '1.1.0-beta01' for compose. Can we fix that please?

-  I changed compose_version = '1.1.0-beta01' to version 1.3.1 in the build.gradle and did a sync. However when I ran the project it wouldn't build. So I need a bit more time to figure this out. 

Can we remove defaults test please?

- Yes, I have done this :+1: 


I will postpone merging until I figure out why I can not use a stable version of compose

File: sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt (hunk 3 above)
Comment (Katie-Marie) [reply sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt]: Hopefully now that RefreshIcon has been added the build will proceed.

File: app/src/main/java/com/sportsworld/learnsharedflow/MainActivity.kt (hunk 1 above)
Comment (Katie-Marie) [reply app/src/main/java/com/sportsworld/learnsharedflow/MainActivity.kt:17]: Hopefully now that RefreshIcon has been added it won't break here.

**Comment by Katie-Marie** [comment]
Project now builds with a stable compose version.

**Comment by Katie-Marie** [comment]
Just another comment on the question @jfsugrue asked above about this code in the composable
```
            messages.add(0, it)
            // Todo: Learn more about recomposition so I can find a better way to achieve this
            messages.removeRange(1,messages.size)
```
            
   The word I couldn't remember in the interview was _intermediaries_. I meant to convey that using an intermediary in the viewModel would have been a better place to do this (as opposed to adapting the flow in the consumable). 
   The section **Modifying the Stream** in the docs explains it well:
   https://developer.android.com/kotlin/flow#modify

**Pull request merged**